packages/metal-dom/test/dom.js

```diff
@@ -625,6 +625,148 @@ describe('dom', function() {
 				dom.triggerEvent(matchedElements[1], 'click');
 				assert.ok(!listener.args[0][0].delegateTarget);
 			});
+
+			it('should not trigger delegate click event for disabled elements', function() {
+				var parent = document.createElement('div');
+				document.body.appendChild(parent);
+
+				var input = document.createElement('input');
+				var select = document.createElement('select');
+				var button = document.createElement('button');
+				var textarea = document.createElement('textarea');
+				input.disabled = true;
+				select.disabled = true;
+				button.disabled = true;
+				textarea.disabled = true;
+				parent.appendChild(input);
+				parent.appendChild(select);
+				parent.appendChild(button);
+				parent.appendChild(textarea);
+
+				var listener1 = sinon.stub();
+				var listener2 = sinon.stub();
+				var listener3 = sinon.stub();
+				var listener4 = sinon.stub();
+
+				dom.delegate(parent, 'click', input, listener1);
+				dom.delegate(parent, 'click', select, listener2);
+				dom.delegate(parent, 'click', button, listener3);
+				dom.delegate(parent, 'click', button, listener4);
+
+				dom.triggerEvent(input, 'click');
+				dom.triggerEvent(select, 'click');
+				dom.triggerEvent(button, 'click');
+				dom.triggerEvent(textarea, 'click');
+				assert.strictEqual(0, listener1.callCount);
+				assert.strictEqual(0, listener2.callCount);
+				assert.strictEqual(0, listener3.callCount);
+				assert.strictEqual(0, listener4.callCount);
+			});
+
+			it('should not trigger delegate click event to an element which its valid parent is disabled', function() {
+				var parent = document.createElement('fieldset');
+				parent.disabled = true;
+				document.body.appendChild(parent);
+
+				var input = document.createElement('input');
+				var select = document.createElement('select');
+				var button = document.createElement('button');
+				var textarea = document.createElement('textarea');
+				parent.appendChild(input);
+				parent.appendChild(select);
+				parent.appendChild(button);
+				parent.appendChild(textarea);
+
+				var listener1 = sinon.stub();
+				var listener2 = sinon.stub();
+				var listener3 = sinon.stub();
+				var listener4 = sinon.stub();
+
+				dom.delegate(parent, 'click', input, listener1);
+				dom.delegate(parent, 'click', select, listener2);
+				dom.delegate(parent, 'click', button, listener3);
+				dom.delegate(parent, 'click', textarea, listener4);
+
+				dom.triggerEvent(input, 'click');
+				dom.triggerEvent(select, 'click');
+				dom.triggerEvent(button, 'click');
+				assert.strictEqual(0, listener1.callCount);
+				assert.strictEqual(0, listener2.callCount);
+				assert.strictEqual(0, listener3.callCount);
+				assert.strictEqual(0, listener4.callCount);
+			});
+
+			it('should trigger delegate click event to an element which its valid parent is disabled unless it is wrapped by an legend element', function() {
```

Comment:
It seems like this description is saying the opposite of what it means. It's saying that the event should trigger unless wrapped by legend, when it's the opposite actually.

That probably happened because the description got confusing. Something like this would work a bit better, for example: `it should trigger delegate click event on elements wrapped by a "legend" tag and with disabled parent`.